app/controllers/placements/support/providers_controller.rb

```diff
@@ -15,14 +15,31 @@ def create
   def check
     @provider = Provider.new(provider_code: params[:accredited_provider_id])
     @provider_details = AccreditedProviderApi.call(@provider.provider_code)
-    return if @provider.valid? && @provider_details.present?
-
-    render :new
+    if @provider.valid? && @provider_details.present?
+      provider_address(@provider_details)
+    else
+      render :new
+    end
   end
 
   private
 
   def provider_params
     params.require(:provider).permit(:provider_code)
   end
+
+  def provider_address(provider_details)
+    @provider_address ||= begin
+      address_parts = [
+        provider_details.dig("attributes", "street_address_1"),
+        provider_details.dig("attributes", "street_address_2"),
+        provider_details.dig("attributes", "street_address_3"),
+        provider_details.dig("attributes", "city"),
+        provider_details.dig("attributes", "county"),
+        provider_details.dig("attributes", "postcode"),
+      ]
+      address_parts.reject!(&:blank?)
+      address_parts.join("<br/>").html_safe
+    end
```

Comment:
I'm not really keen on this being in the controller -- it seems like the work of a decorator / presenter. Or maybe a helper. but perhaps that doesn't work so well with this not being an ActiveRecord?

Reply:
As discussed, because this information is fed from the API as `JSON` I didn't know where to put this.
Given the Ollie is working on Importing this data into a Provider model this should become redundant/replaceable fairly soon.